javalib/src/main/scala/java/lang/Math.scala

```diff
@@ -126,8 +138,52 @@ object Math {
     }
 
   // Wasm intrinsics
-  @inline def ceil(a: scala.Double): scala.Double = js.Math.ceil(a)
-  @inline def floor(a: scala.Double): scala.Double = js.Math.floor(a)
+  @inline def ceil(a: scala.Double): scala.Double =
+    linkTimeIf(LinkingInfo.targetPureWasm) {
+      -floor(-a)
+    } {
+      js.Math.ceil(a)
+    }
+  @inline def floor(a: scala.Double): scala.Double = {
+    linkTimeIf(LinkingInfo.targetPureWasm) {
+      // Ported from https://www.netlib.org/fdlibm/s_floor.c
+      var bits = Double.doubleToRawLongBits(a)
```

Comment:
For consistency (and arguably readability), put the pure-Wasm implementation in a separate `private def`?